orchestrator/packages/api/src/controllers/grading-queue-controller.ts

```diff
@@ -175,3 +176,22 @@ export const deleteJob = async (req: Request, res: Response) => {
     }
   }
 };
+
+export const getJobStatus = async (req: Request, res: Response) => {
+  const { key, response_url } = req.body;
+  const jobQueueStatus = await getJobQueueStatus(key, response_url);
+  if (!jobQueueStatus) {
+    return res.json("We could not find the job you're looking for. Please contact a professor or admin.");
+  }
+  const { reservation, numReservationsAhead } = jobQueueStatus;
+  const now = new Date();
+  if (now > reservation.releaseAt) {
```

Comment:
This seems like a weird place for this logic, since surely it's present somewhere else too?

Reply:
Hasn't been relevant yet; added a TODO to go back ad replace that once the abstraction is to be created.